src/fides/api/migrations/post_upgrade_index_creation.py

```diff
@@ -0,0 +1,251 @@
+            "type": "constraint",
+        },
+        # column indexes
+        {
+            "name": "ix_staged_resource_ancestor_ancestor",
+            "statement": "CREATE INDEX CONCURRENTLY ix_staged_resource_ancestor_ancestor ON stagedresourceancestor (ancestor_urn)",
+            "type": "index",
+        },
+        {
+            "name": "ix_staged_resource_ancestor_descendant",
+            "statement": "CREATE INDEX CONCURRENTLY ix_staged_resource_ancestor_descendant ON stagedresourceancestor (descendant_urn)",
+            "type": "index",
+        },
+    ],
+}
+
+
+def check_object_exists(db: Session, object_name: str) -> bool:
+    """Checks Postgres' system catalogs to verify the existence of a given index or constraint in a specific table."""
+    object_query = text(
+        """
+        SELECT EXISTS (
+            SELECT 1
+            FROM pg_indexes
+            WHERE indexname = :object_name
+        ) OR EXISTS (
+            SELECT 1
+            FROM pg_constraint
+            WHERE conname = :object_name
+        )
+        """
+    )
+    result = db.execute(
+        object_query,
+        {
+            "object_name": object_name,
+        },
+    ).scalar()
+    return result
+
+
+def create_object(db: Session, object_statement: str, object_name: str) -> None:
+    """Executes the index or constraint creation statement."""
+    logger.info(f"Creating index/constraint object by executing: '{object_name}'...")
+    with db.bind.connect().execution_options(
+        isolation_level="AUTOCOMMIT"
+    ) as connection:
+        connection.execute(text(object_statement))
+    logger.info(f"Successfully created index/constraint object: '{object_name}'")
+
+
+def check_and_create_objects(
+    db: Session, table_object_map: Dict[str, List[Dict[str, str]]]
+) -> Dict[str, str]:
+    """Returns a dictionary of any indices or constraints that are in the process of being created."""
+    object_info: Dict[str, str] = {}
+    for _, objects in table_object_map.items():
+        for object_data in objects:
+            object_name = object_data["name"]
+            object_statement = object_data["statement"]
+            object_type = object_data["type"]
+            object_exists = check_object_exists(db, object_name)
+
+            if not object_exists:
+                if object_type == "index":
+                    constraint_name = object_data.get("constraint_name")
+                    if constraint_name:
+                        constraint_exists = check_object_exists(db, constraint_name)
+                        if constraint_exists:
+                            logger.debug(
+                                f"Constraint {constraint_name} already exists, skipping index creation for {object_name}"
+                            )
+                            continue
+
+                create_object(db, object_statement, object_name)
```

Comment:
@galvana noting here - we need to think about concurrent execution because it doesn't seem like this is quite threadsafe given some testing...